scripts/dts/gen_defines.py

```diff
@@ -370,15 +373,15 @@ def write_ranges(node):
     for i,range in enumerate(node.ranges):
         idx_vals.append((f"{path_id}_RANGES_IDX_{i}_EXISTS", 1))
 
-        if node.bus == "pcie":
+        if node.buses is not None and "pcie" in node.buses:
```

Comment:
Can you use an empty list when there are no buses? That way you won't need the null check. The cases are:

- no bus: `node.buses` is []
- one bus: `node.buses` has one element, e.g. ["pcie"]
- multiple buses: `node.buses` has multiple elements, e.g. ["i2c", "i3c"]

Reply:
Changed to do empty list instead of None.